doc/pveclibmaindox.h

```diff
@@ -414,6 +416,93 @@
+*  (<B>LE</B>) data format in the Linux ecosystem.  The goal was to
+*  enhance application code portability across Linux platforms.
+*  This goal was integrated into the OpenPOWER ELF V2
+*  Application Binary Interface <B>ABI</B> specification.
+*
+*  The POWER8 processor architecturally supports an <I>Endian Mode</I>
+*  and supports both BE and LE storage access in hardware.
+*  However register to register operations do not change
+*  (are not effected by endian mode).  The ABI extends the LE
+*  storage format to vector register (logical)
+*  element numbering.
+*
+*  This has no effect for most altivec.h
+*  operations where the input elements and the results "stay in their
+*  lanes".  For operations of the form (T[n] = A[n] op B[n]),
+*  it does not matter if elements are numbered [0, 1, 2, 3] or
+*  [3, 2, 1, 0].
+*
+*  But there are cases where element renumbering can change the
+*  results.  Changing element numbering does change the even / odd
+*  relationship for merge and integer multiply.  For <B>LE</B>
+*  targets, operations accessing even vector elements are
+*  implemented using the equivalent odd instruction (and visa versa)
+*  and inputs are swapped.  Similarly for high and low merges.
+*  Inputs are also swapped for Pack, Unpack, and Permute operations
+*  and the permute select vector is inverted.
+*  The above is just a sampling of a larger list <I>LE transforms</I>.
+*  The OpenPOWER ABI specification provides a helpful table of
+*  <a href="http://openpowerfoundation.org/wp-content/uploads/resources/leabi/content/dbdoclet.50655244_90667.html"
+*  >Endian-Sensitive Operations</a>.
+*
+*  \note This means that the vector built-ins provided by altivec.h
+*  may not generate the instructions you expect.
+*
+*  This does matter when doing extended precision arithmetic.
+*  Here we need to maintain most-to-least significant byte order and
+*  align "digit" columns for summing partial products.  Many of these
+*  operation where defined long before Little Endian was seriously
+*  considered and are decidedly Big Endian in register format.
+*  Basically any operation where the element
+*  size changes from input to output is suspect for <B>LE</B>
+*  targets.
+*
+*  The coding for these higher level operations
+*  is complicated by <I>Little Endian</I> (LE) support as
+*  specified in the OpenPOWER ABI and as implemented in the compilers.
+*  Little Endian changes the effective vector element numbering and
+*  the location of even and odd elements.
+*
+*  This is a general problem for using vectors to implement extended
+*  precision arithmetic.
+*  The multiply even/odd operations being the primary example.
+*  The products are double-wide and in BE order in the vector register.
+*  This is reinforced by the Vector Add/Subtract Unsigned Quadword
+*  instructions.
+*  And the products from even multiply instruction are always
+*  <I>algebraically</I> higher then the odd products.
```

Comment:
(my ignorance on display), but what is meant by 'even multiply products are algebraically higher than odd products?